### ISSUE TITLE
dev: explore removing `transaction_hash` field from `EventWrapper`

### ISSUE THREAD

**Issue by greged93** [keep-starknet-strange/madara#510]
In order to more easily pass the transaction hash for an event into the RPC, a `transaction_hash` field was added to the `EventWrapper` structure. Explore an improvement which would allow to remove this field and still obtain the transaction hash for the events in the RPC method `get_events`.

**Comment by LucasLvy** [comment]
Any news ?

**Comment by nikania** [comment]
I can take this

**Comment by cwkang1998** [comment]
> Any news ?

Sorry @LucasLvy , missed your message. Its taking me some time on this but @nikania feel free to take it. 

So far I've essentially tried to extract `transaction_hash` into a `EventContext`, but the `transaction_hash` will still need to pass through the function call. Is that what you want to achieve or am I misunderstanding it?

**Comment by cwkang1998** [comment]
@nikania Opened a draft PR on the work I am doing in this https://github.com/keep-starknet-strange/madara/pull/661.
Currently have not extracted `transaction_hash` out into a `EventContext` fully yet, need to add tests and fix some issues with the traits. Hope it might be helpful to you.

**Comment by nikania** [comment]
Created draft PR, where I changed the mechanism of getting events from the rpc endpoint getEvents by taking events from transaction receipts. Changing pagination was cumbersome🤯, so I added 2 integration tests while debugging. 
(I also have a suggestion to change the pagination mechanism in this case, for example, changing the `continuation_token` to tuple `continuation_token = (trx_receipt_index, event_index)` - or maybe even `continuation_token = (block_index, trx_receipt_index, event_index)` to make skipping already received events easier)

This change eliminates the need for the `transaction_hash` field in `EventWrapper` structure.
So, if you approve it - the next step I can just remove the above-mentioned field.
